dataRetriever/txpool/mempoolHost.go

```diff
@@ -0,0 +1,112 @@
+package txpool
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/process"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+	"github.com/multiversx/mx-chain-vm-common-go/parsers"
+)
+
+type argsMempoolHost struct {
+	txGasHandler txGasHandler
+	marshalizer  marshal.Marshalizer
+}
+
+type mempoolHost struct {
+	txGasHandler        txGasHandler
+	callArgumentsParser process.CallArgumentsParser
+	esdtTransferParser  vmcommon.ESDTTransferParser
+}
+
+func newMempoolHost(args argsMempoolHost) (*mempoolHost, error) {
+	if check.IfNil(args.txGasHandler) {
+		return nil, dataRetriever.ErrNilTxGasHandler
+	}
+	if check.IfNil(args.marshalizer) {
+		return nil, dataRetriever.ErrNilMarshalizer
+	}
+
+	argsParser := parsers.NewCallArgsParser()
+
+	esdtTransferParser, err := parsers.NewESDTTransferParser(args.marshalizer)
+	if err != nil {
+		return nil, err
+	}
+
+	return &mempoolHost{
+		txGasHandler:        args.txGasHandler,
+		callArgumentsParser: argsParser,
+		esdtTransferParser:  esdtTransferParser,
+	}, nil
+}
+
+// ComputeTxFee computes the fee for a transaction.
+func (host *mempoolHost) ComputeTxFee(tx data.TransactionWithFeeHandler) *big.Int {
+	return host.txGasHandler.ComputeTxFee(tx)
+}
+
+// GetTransferredValue returns the value transferred by a transaction.
+func (host *mempoolHost) GetTransferredValue(tx data.TransactionHandler) *big.Int {
+	value := tx.GetValue()
+	hasValue := value != nil && value.Sign() != 0
+	if hasValue {
+		// Early exit (optimization): a transaction can either bear a regular value or be a "MultiESDTNFTTransfer".
+		return value
+	}
+
+	data := tx.GetData()
+	hasData := len(data) > 0
+	if !hasData {
+		// Early exit (optimization): no "MultiESDTNFTTransfer" to parse.
+		return tx.GetValue()
+	}
+
+	maybeMultiTransfer := bytes.HasPrefix(data, []byte(core.BuiltInFunctionMultiESDTNFTTransfer))
+	if !maybeMultiTransfer {
+		// Early exit (optimization).
+		return nil
```

Comment:
wouldn't it be easier/safer to return big.NewInt(0) here? (and L79, L84, L88)

Reply:
Fixed.